From 00983b5b1d3c5451f8941d4c7258e8a0037a6738 Mon Sep 17 00:00:00 2001
From: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date: Wed, 23 Nov 2016 15:16:56 +0300
Subject: [PATCH] PORTAL-2440 Add application key for metadata update in case
 it does not exist Reviewed By: Pedro P. Candel <pedro@joyent.com>

---
 lib/index.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/index.js b/lib/index.js
index b4cfd2d..7904ddd 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2292,6 +2292,9 @@ function modifyMetadata(user, key, md, account, cb) {
 
         self.getMetadata(user, key, account, function (err) {
             if (err && err.statusCode === 404) {
+                if (!md.cn) {
+                    md.cn = key;
+                }
                 self.add(dn, md, function (err2) {
                     if (err2) {
                         cb(translateError(err2));
-- 
2.21.0

